src/libcmd/command.cc

```diff
@@ -37,11 +38,12 @@ nlohmann::json NixMultiCommand::toJSON()
 void NixMultiCommand::run()
 {
     if (!command) {
-        std::set<std::string> subCommandNames;
+        std::set<std::string> subCommandTextLines;
         for (auto & [name, _] : commands)
-            subCommandNames.insert(name);
-        throw UsageError("'nix %s' requires a sub-command.\nAvailable sub-commands: %s",
-                commandName, concatStringsSep(", ", subCommandNames));
+            subCommandTextLines.insert(fmt("- `%s`", name));
+        std::string markdownError = fmt("`nix %s` requires a sub-command.\n\nAvailable sub-commands:\n\n%s\n",
```

Comment:
```suggestion
        std::string markdownError = fmt("`nix %s` requires a sub-command. Available sub-commands:\n\n%s\n",
```

This may take too much vertical space otherwise. We need that for the list. Just a non-blocking thought.

Reply:
Indeed, that's better without the extra newlines, thanks :+1: